packages/back-end/src/models/ExperimentModel.ts

```diff
@@ -190,6 +563,181 @@ export const logExperimentUpdated = async ({
+ */
+export async function deleteExperimentByIdForOrganization(
+  id: string,
+  organization: OrganizationInterface
+) {
+  try {
+    const previous = await findExperiment({
+      experimentId: id,
+      organizationId: organization.id,
+    });
+    if (previous) {
+      await logExperimentDeleted(organization, previous);
+    }
+  } catch (e) {
+    logger.error(e);
+  }
+
+  await ExperimentModel.deleteOne({
+    id,
+  });
+}
+
+/**
+ * Removes the tag from any experiments that have it
+ * and logs the experiment.updated event
+ * @param organization
+ * @param tag
+ */
+export const removeTagFromExperiments = async ({
+  organization,
+  tag,
+}: {
+  organization: OrganizationInterface;
+  tag: string;
+}): Promise<void> => {
+  const query = { organization: organization.id, tags: tag };
+  const previousExperiments = await ExperimentModel.find(query);
+
+  await ExperimentModel.updateMany(query, {
+    $pull: { tags: tag },
+  });
+
+  previousExperiments.forEach((previous) => {
+    const current = cloneDeep(previous);
+    current.tags = current.tags.filter((t) => t != tag);
+
+    logExperimentUpdated({
+      organization,
+      previous,
+      current,
+    });
+  });
+};
+
+export async function getExperimentsByOrganization(
+  organization: string,
+  project?: string
+) {
+  const query: FilterQuery<ExperimentDocument> = {
+    organization,
+  };
+
+  if (project) {
+    query.project = project;
+  }
+
+  const experiments = await ExperimentModel.find(query);
+
+  return experiments.map(toInterface);
+}
+
+export async function removeMetricFromExperiments(
+  metricId: string,
+  organization: OrganizationInterface
+) {
+  const oldExperiments: Record<
+    string,
+    {
+      previous: ExperimentInterface | null;
+      current: ExperimentInterface | null;
+    }
+  > = {};
+
+  const orgId = organization.id;
+
+  const metricQuery = { organization: orgId, metrics: metricId };
+  const guardRailsQuery = { organization: orgId, guardrails: metricId };
+  const activationMetricQuery = {
+    organization: orgId,
+    activationMetric: metricId,
+  };
+  const docsToTrackChanges = await ExperimentModel.find({
+    $or: [metricQuery, guardRailsQuery, activationMetricQuery],
+  });
+  docsToTrackChanges.forEach((experiment: ExperimentDocument) => {
+    if (!oldExperiments[experiment.id]) {
+      oldExperiments[experiment.id] = {
+        previous: experiment,
+        current: null,
+      };
+    }
+  });
+
+  // Remove from metrics
+  await ExperimentModel.updateMany(metricQuery, {
+    $pull: { metrics: metricId },
+  });
+
+  // Remove from guardrails
+  await ExperimentModel.updateMany(guardRailsQuery, {
+    $pull: { guardrails: metricId },
+  });
+
+  // Remove from activationMetric
+  await ExperimentModel.updateMany(activationMetricQuery, {
+    $set: { activationMetric: "" },
+  });
+
+  const ids = Object.keys(oldExperiments);
+  const updatedExperiments = await ExperimentModel.find({ id: { $in: ids } });
```

Comment:
Need to use org Id in the query and use toInterface on the result. Also, we might not even need this query. We already have all the experiments in memory and know how they changed. We can use the same cloneDeep method we use elsewhere

Reply:
I've updated the query for now - at our 1:1 today let's talk through the alternate approach of using `cloneDeep`.